### ISSUE TITLE
Inconsistent result observed: same input leads to different results

### ISSUE THREAD

**Issue by starrify** [scrapinghub/dateparser#358]
Version: 1c54cea

An example to reproduce the issue:
```
(env2) [pengyu@GLaDOS-Precision-7510 dateparser]$ python
Python 2.7.14 (default, Sep 20 2017, 01:25:59) 
[GCC 7.2.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import dateparser
>>> text_a = u'10月'
>>> text_b = u'礼拜日'
>>> dateparser.parse(text_a)
datetime.datetime(2017, 10, 31, 0, 0)
>>> dateparser.parse(text_b)
datetime.datetime(2017, 10, 29, 0, 0)
>>> dateparser.parse(text_a)
datetime.datetime(2017, 1, 1, 6, 2, 22, 912807)
```

I've not yet got any further test, but there seems to be some internal state that is incorrectly managed.

**Comment by nurlansf** [comment]
Saw same behaviour when running dateparser.parse() in multiple threads. Was solved by running it using multiple processes. Library doesn't seem to be thread safe.

**Comment by 0xnurl** [comment]
Saw same behaviour when running dateparser.parse() in multiple threads. Was solved by running it using multiple processes. Library doesn't seem to be thread safe.

**Comment by srhrshr** [comment]
@manycoding, I believe this issue has been addressed in the `0.7.0` release as well - can be marked as resolved and closed. 

```
>>> text_a = u'10月'

>>> text_b = u'礼拜日'

>>> dateparser.parse(text_a)
datetime.datetime(2019, 10, 22, 0, 0)

>>> dateparser.parse(text_b)
datetime.datetime(2019, 10, 20, 0, 0)

>>> for t in [text_a] * 100:
...     assert datetime.datetime(2019, 10, 22, 0, 0) == dateparser.parse(text_a)

>>> for t in [text_b] * 100:
...     assert datetime.datetime(2019, 10, 20, 0, 0) == dateparser.parse(text_b)

```

**Comment by manycoding** [comment]
Fixed in `0.7.2`
